File: TME.Store/TME.Store.Domain/Models/ProductPrice.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;

namespace TME.Store.Domain.Models
{
    public class ProductPrice
    {
        public int VatRate { get; private set; }
        public string VatType { get; private set; }
        public float MinPrice { get; private set; }
        public IReadOnlyCollection<PriceOffer> PriceOffers { get; private set; }

        public ProductPrice(int vatRate,
                            string vatType,
                            IList<PriceOffer> offers)
        {
            VatRate = vatRate;
            VatType = vatType;
            
            PriceOffers = new ReadOnlyCollection<PriceOffer>(offers);
            MinPrice = PriceOffers.Min(offer => offer.Price);
        }
    }
}

File: tme_store/lib/models/product_price.dart
import 'dart:math';

import 'package:tme_store/models/price_orffer.dart';


class ProductPrice {
  final int vatRate;
  final String vatType;
  double _minPrice;
  double get minPrice => _minPrice;
  final List<PriceOffer> priceOffers;

  ProductPrice({
    this.vatRate,
    this.vatType,
    this.priceOffers
  }){
    _minPrice = priceOffers.map((po) => po.price).reduce(min);
  }
}